File: FMAplication/ClientApp/src/app/Shared/Entity/Daily-Activity/survey-report.ts
import { Question } from '..';
import { SurveyQuestionSet } from '../Questions';
import { DailyCMActivity } from './daily-cm-activity';

export class SurveyReport {

    public id : number;
    public dailyCMActivityId : number;
    public questionId : number;
    public answer : string;
    public surveyId : number;
    public isConsumerSurvey : boolean;

    public status: number;

    public question: Question;
    public survey : SurveyQuestionSet;
    public dailyCMActivity: DailyCMActivity;

    // display

    public displayStatus: string;
    public displayIsConsumerSurvey: string;
    public surveyName : string;
    public questionName : string;

    public displayDate : string;
    public fmUserName : string;
    public cmUserName : string;
    public outletName : string;
    public salesPointName : string;
}

File: FMAplication/Domain/Reports/SurveyReport.cs
using FMAplication.Core;
using FMAplication.Domain.DailyActivities;
using FMAplication.Domain.Examples;
using FMAplication.Domain.Questions;
using FMAplication.Enumerations;
using System.ComponentModel.DataAnnotations;

namespace FMAplication.Domain.Reports
{

    public class SurveyReport : AuditableEntity<int>
    {
        public int DailyCMActivityId { get; set; }
        public int QuestionId { get; set; }
        public string Answer { get; set; }
        public int SurveyId { get; set; }
        public Question Question { get; set; }
        public SurveyQuestionSet Survey { get; set; }
        public DailyCMActivity DailyCMActivity { get; set; }
        public bool IsConsumerSurvey { get; set; }
    }
}
